src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,58 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.OrderDao;
+import com.dev.cinema.exception.DataProcessingException;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.Order;
+import com.dev.cinema.model.User;
+import com.dev.cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
+    @Override
+    public Order add(Order order) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(order);
+            transaction.commit();
+            return order;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException(
+                    "Cant add order with user id: " + order.getUser().getId()
+                            + " to the database", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Order> getUserOrders(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Order> getUserOrdersQuery = session.createQuery(
+                    "SELECT DISTINCT o FROM Order o "
+                            + "JOIN FETCH o.user "
+                            + "JOIN FETCH o.tickets t "
+                            + "JOIN FETCH t.movieSession ms "
+                            + "JOIN FETCH ms.movie "
+                            + "JOIN FETCH ms.cinemaHall "
+                            + "WHERE o.user.id = :userId", Order.class)
+                    .setParameter("userId", user.getId());
+            return getUserOrdersQuery.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException(
+                    "Can't get orders of user with id: " + user.getId(), e);
+        }
```

Comment:
Is this catch block necessary?

Reply:
It helps you quickly identify where the problem is